### PR TITLE
credentials: set precise expire times

### PR DESCRIPTION
**Pull request by pjonsson**
Use the exp/iat times in the token
to compute precise expire times.

### CODE REVIEW

File: src/cdsetool/credentials.py [hunk 1]
```diff
@@ -170,17 +168,22 @@ def __ensure_tokens(self):
...
                 raise TokenClientConnectionError from e
-            jwt.decode(
+            data = jwt.decode(
                 self.__access_token,
                 key=key.key,
                 algorithms=key._algorithms,  # pylint: disable=protected-access
                 options={"verify_aud": False},
             )
+            if refresh_expire_delta is not None:
+                self.__access_token_expires = datetime(*localtime(data["exp"])[:6])
```
Comment (pedep) [inline src/cdsetool/credentials.py:183]:
```suggestion
                self.__access_token_expires = datetime.fromtimestamp(data["exp"])
```

Comment (pedep) [reply src/cdsetool/credentials.py:183]:
```python3
ts = 1715611926
assert datetime(*localtime(ts)[:6]) == datetime.fromtimestamp(ts)

File: src/cdsetool/credentials.py [hunk 2]
```diff
@@ -170,17 +168,22 @@ def __ensure_tokens(self):
...
+                self.__refresh_token_expires = (
+                    datetime(*localtime(data["iat"])[:6]) + refresh_expire_delta
```
Comment (pedep) [inline src/cdsetool/credentials.py:185]:
```suggestion
                    datetime.fromtimestamp(data["iat"]) + refresh_expire_delta
```

**Review by pedep — CHANGES_REQUESTED** [review]

File: src/cdsetool/credentials.py (hunk 1 above)
Comment (pjonsson) [reply src/cdsetool/credentials.py]: Much better, thanks! Fixed!

File: src/cdsetool/credentials.py (hunk 2 above)
Comment (pjonsson) [reply src/cdsetool/credentials.py]: Fixed.

**Review by pedep — APPROVED** [review]

**Pull request merged**